chain/epoch_manager/src/lib.rs

```diff
@@ -1084,19 +1085,22 @@ impl EpochManager {
                 )
             }
             ValidatorInfoIdentifier::BlockHash(ref h) => {
-                let aggregator = self.get_and_update_epoch_info_aggregator(&epoch_id, h, true)?;
+                // If we are here, `h` is hash of the latest block of the
+                // current epoch.
+                self.update_epoch_info_aggregator(&epoch_id, h, None)?;
```

Comment:
I think it's fine to use `self.epoch_info_aggregator` directly here, instead of calling `update_epoch_info_aggregator`, since we are always accessing the latest block, and after your change `epoch_info_aggregator` will always be updated to chain head. That way, we can also change the `ValidatorInfoIdentifier::BlockHash` option to `ValidatorInfoIdentifier::LatestBlock`.

Reply:
> That way, we can also change the ValidatorInfoIdentifier::BlockHash option to ValidatorInfoIdentifier::LatestBlock.

So this is something I'm not clear about.  Normally we call the update on final blocks (https://github.com/near/nearcore/blob/master/chain/epoch_manager/src/lib.rs#L602) which is a few blocks behind latest block.  When we’re called with ValidatorInfoIdentifier::BlockHash we’re called with a block which hasn’t been aggregated yet, no?

Reply:
Ah great observation, I didn't notice that. I think it's fine to use the latest final block, since right now the only use cases for get_validator_info using block hashes are in log_summary and in GetValidatorInfo, either cases, I don't think the distinction matters that much and I'd choose to keep the code simpler and cleaner over getting these two cases really accurate. We could use `ValidatorInfoIndentifier::LastestFinalBlock` to signal the correct meaning.

Reply:
Let’s leave it as `ValidatorInfoIndentifier::BlockHash` for now. I’ll look into simplifying it in another PR.